test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/MonthDayTest.scala

```diff
@@ -0,0 +1,218 @@
+package org.scalajs.testsuite.javalib.time
+
+import java.time.chrono.IsoChronology
+import java.time.temporal.{UnsupportedTemporalTypeException, ValueRange, ChronoField}
+import java.time.{DateTimeException, LocalDate, Month, MonthDay}
+
+import org.junit.Assert._
+import org.junit.Test
+import org.scalajs.testsuite.utils.AssertThrows._
+
+/**
+  * Created by alonsodomin on 22/12/2015.
+  */
+class MonthDayTest extends TemporalAccessorTest {
+  import ChronoField._
+  import MonthDay._
+  import DateTimeTestUtil._
+
+  final val min = of(Month.JANUARY, 1)
+  final val max = of(Month.DECEMBER, 31)
+  final val leapMonth = of(Month.FEBRUARY, 29)
+
+  val samples = Month.values().map(month => (month, month.minLength(), month.maxLength())).flatMap {
+    case (month, minDay, maxDay) =>
+      if (minDay != maxDay) Seq(of(month, 1), of(month, minDay), of(month, maxDay))
+      else Seq(of(month, 1), of(month, minDay))
+  }.toSeq
+
+  val yearSamples = 2000 to 2020
+  val leapYearLifted = (year: Int) => IsoChronology.INSTANCE.isLeapYear(year)
+
+  override def isSupported(field: ChronoField): Boolean =
+    field == ChronoField.MONTH_OF_YEAR || field == ChronoField.DAY_OF_MONTH
+
+  @Test override def test_range(): Unit = {
```

Comment:
Well, I had to override the `range` method defined in `TemporalAccessor` because the test was originally failing when testing the output of that `range` method against the JDK MonthDay class.

The scenario is, the default `ValueRange` for the `TemporalField` `DAY_OF_MONTH` gives a minimum value of `1`, a minimum length of `28` and a maximum length of `31`. So the `range` method in `TemporalAccessor` will return always that value range regardless of the actual month value stored in the `TemporalAccessor` concrete implementation.

This means that when asked for the `DAY_OF_MONTH` range of a `MonthDay` representing any day of January, the `range` method will return `ValueRange.of(1, 28, 31)`. Accordingly it will return the same `ValueRange` when asked for the range of that same field for a `MonthDay` representing any day of Feburary.

However the JDK seems to behave differently as the test failed with the first `MonthDay` sample, saying that it was expecting a `ValueRange.of(1, 28, 31)` but it got a `ValueRange.of(1, 31)` instead. Similar behaviour happens with any February sample but in this case, whilst the `test_range` method in the `TemporalAccessorTest` class was expecting a `ValueRange.of(1, 28, 31)` it got a `ValueRange.of(1, 28, 29)`.

Due to this I decided to override the `range` method to make it aware of the actual month represented by the `MonthDay` instance, and this forced me to override the `test_range` method in the test as well.